applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1309,70 +1205,15 @@ int unicast_client_stop(void)
 int unicast_client_send(struct le_audio_encoded_audio enc_audio)
 {
 	int ret;
-	size_t data_size_pr_stream;
-	struct bt_iso_tx_info tx_info = {0};
-	struct sdu_ref_msg msg;
-
-	if ((enc_audio.num_ch == 1) || (enc_audio.num_ch == ARRAY_SIZE(headsets))) {
-		data_size_pr_stream = enc_audio.size / enc_audio.num_ch;
-	} else {
-		LOG_ERR("Num encoded channels must be 1 or equal to num headsets");
-		return -EINVAL;
-	}
+	static struct bt_bap_stream *bap_tx_streams[CONFIG_BT_MAX_CONN];
 
-	if (data_size_pr_stream != LE_AUDIO_SDU_SIZE_OCTETS(CONFIG_BT_AUDIO_BITRATE_UNICAST_SINK)) {
-		LOG_ERR("The encoded data size does not match the SDU size");
-		return -EINVAL;
-	}
-
-	if (ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
-		ret = bt_iso_chan_get_tx_sync(&headsets[AUDIO_CH_L].sink_stream.ep->iso->chan,
-					      &tx_info);
-	} else if (ep_state_check(headsets[AUDIO_CH_R].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
-		ret = bt_iso_chan_get_tx_sync(&headsets[AUDIO_CH_R].sink_stream.ep->iso->chan,
-					      &tx_info);
-	} else {
-		/* This can happen if a headset is reset and the state in
-		 * streamctrl hasn't had time to update yet
-		 */
-		LOG_DBG("No headset in stream state");
-		return -ECANCELED;
+	for (int i = 0; i < CONFIG_BT_MAX_CONN; i++) {
```

Comment:
So we populate this array every time we send data? This array could be global for this file and the pointers should be set when the headsets[i].sink_stream is populated. Same for broadcast_source

It also seems strange to use the BT_MAX_CONN as size since we could in theory have 4 ACLs but only 1 CIS. Suggest using CONFIG_BT_ASCS_ASE_SRC_COUNT instead